content/library/api/mutate.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Animate and update elements
+slug: /develop/concepts/design/animate
+description: st.add_rows appends a dataframe to the bottom of the current one in certain elements, for optimized data updates.
+---
+
+# Animate and update elements
+
+Sometimes you display a chart or dataframe and want to modify live as the app
+runs (for example, in a loop). Some elements have built-in methods to allow you
```

Comment:
We should mention that **almost all** elements that aren't interactive widgets return an object (a DeltaGenerator, but IDK if we want to provide all that detail) that allows you update them in-place.

Reply:
I'll categorize that as follow up. In general, I want to make documentation more complete and that includes talking about DeltaGenerators. However, I think it'd be confusing to just casually mention them on the side. When I get a more proper introduction to the idea, and I get all the API reference material to reflect all those commands that return DeltaGenerators, then it will totally make sense to refer to them where they are relevant.